lib/manageiq/appliance_console/oidc_authentication.rb

```diff
@@ -74,6 +76,7 @@ def validate_oidc_options
         raise "Must specify the OpenID-Connect Provider URL via --oidc-url" if options[:oidc_url].blank?
         raise "Must specify the OpenID-Connect Client ID via --oidc-client-id" if options[:oidc_client_id].blank?
         raise "Must specify the OpenID-Connect Client Secret via --oidc-client-secret" if options[:oidc_client_secret].blank?
+        raise "Must specify the OpenID-Connect Client Introspection Endpoint --oidc-introspection-endpoint" if options[:oidc_introspection_endpoint].blank?
```

Comment:
Why are we making it required?  I would expect that most providers have a well known endpoint, so I would argue this is optional, and if specified then we include the line.  @abellotti Thoughts?

Reply:
Any objections/issues to having the CLI attempt to fetch the provider metadata at the url specified ?  If we can, then we can simply get the exact introspection URL needed, this would work for all providers and not just keycloak (as well as future IAM where the provider metadata would be specified).

so enhanced logic would be:

- if introspection url not specified:
  - try to fetch the provider metadata, if successful, use the introspection value there.
  - if unsuccessful, try to derive as done in this PR.
  - if cannot derive, fail by requiring the option.

Reply:
> Any objections/issues to having the CLI attempt to fetch the provider metadata at the url specified ? If we can, then we can simply get the exact introspection URL needed, this would work for all providers and not just keycloak (as well as future IAM where the provider metadata would be specified).
> 
> so enhanced logic would be:
> 
> * if introspection url not specified:
>   
>   * try to fetch the provider metadata, if successful, use the introspection value there.
>   * if unsuccessful, try to derive as done in this PR.
>   * if cannot derive, fail by requiring the option.

Are you suggesting something like this:

```bash
curl -k -L -X GET     -H 'Content-Type: application/x-www-form-urlencoded'     -d 'grant_type=password'     -d 'client_id=${options[:oidc_client_id]}'     -d 'client_secret=${options[:oidc_client_secret]}'  ${options[:oidc_url]}  | sed -e 's/[{}]/''/g'  | sed -s 's/.*introspection_endpoint\":\"//' | sed -s 's/.\".*$//'
```

I do not think we should rely on something like `jq` being there to parse the JSON.

I'm a little concerned that parsing the JSON might be more trouble than the outcome would be worth.

Thoughts?